docs/the-new-architecture/turbo-modules-with-swift.md

```diff
@@ -0,0 +1,167 @@
+:::
+
+This guide starts from the iOS implementation of the [Native Module](/docs/next/turbo-native-modules-introduction) guide.
+Make sure to be familiar with that guide before diving into this one, potentially implementing the example in the guide.
+
+## The Adapter pattern
+
+The goal we have is to implement all our business logic using a Swift module and have a thin glue layer in Objective-C++ that is able to connect the app with the Swift implementation.
+
+We can achieve this by leveraging the [Adapter](https://refactoring.guru/design-patterns/adapter) design pattern, to connect the Swift Module with the Objective-C++ layer.
+
+The Objective-C++ object is created by React Native and it keeps a reference to the Swift module, handling its lifecycle. The Objective-C++ object forwards to the all the methods invocation to Swift.
+
+### Creating the Swift Module
+
+The first step is to move the implementation from the Objective-C++ layer to the Swift Layer.
+
+To achieve that, please follow these steps:
+
+1. Create a new empty file in the Xcode project, and call it `NativeLocalStorage.swift`
+2. Add the implementation in your Swift module like it follows:
+
+```swift title="NativeLocalStorage.swift"
+import Foundation
+
+@objc public class NativeLocalStorage: NSObject {
+  let userDefaults = UserDefaults(suiteName: "local-storage");
+
+
+  @objc public func getItem(for key: String) -> String? {
+    return userDefaults?.string(forKey: key)
+  }
+
+  @objc public func setItem(for key: String, value: String) {
+    userDefaults?.set(value, forKey: key)
+  }
+
+  @objc public func removeItem(for key: String) {
+    userDefaults?.removeObject(forKey: key)
+  }
+
+  @objc public  func clear() {
+    userDefaults?.dictionaryRepresentation().keys.forEach { removeItem(for: $0) }
+  }
+}
+
+```
+
+Notice that you have to declare all the methods that you need to call from Objective-C as `public` and with the `@objc` annotation.
+Remember also to make your class inherit from `NSObject`, otherwise it would not be possible to use it from Objective-C.
+
+### Update the `RCTNativeLocalStorage` file
+
+Then, we need to update the implementation of the `RCTNativeLocalStorage` to be able to create the Swift module and to call its methods.
+
+1. Open the `RCTNativeLocalStorage.mm` file
+2. Update it as it follows:
+
+```diff title="RCTNativeLocalStorage.mm"
+//  RCTNativeLocalStorage.m
+//  TurboModuleExample
+
+#import "RCTNativeLocalStorage.h"
++#import "SampleApp-Swift.h"
+
+- static NSString *const RCTNativeLocalStorageKey = @"local-storage";
+
+-@interface RCTNativeLocalStorage()
+-@property (strong, nonatomic) NSUserDefaults *localStorage;
+-@end
+
+-@implementation RCTNativeLocalStorage
++@implementation RCTNativeLocalStorage {
++    NativeLocalStorage *storage;
++}
+
+-RCT_EXPORT_MODULE(NativeLocalStorage)
```

Comment:
It might be worth a callout that this line in particular is not / no longer needed